src/main/java/org/jenkinsci/tracey/RabbitMQTrigger.java

```diff
@@ -113,33 +140,36 @@ private TraceyRabbitMQBrokerImpl configureBroker(Job<?,?> proj, String hid) {
         } catch (InterruptedException ex) {
             LOG.log(Level.SEVERE, "InterruptedException caught", ex);
         }
-        if(upw != null) {
-            broker = new TraceyRabbitMQBrokerImpl(new RabbitMQConnection(env.expand(rHost),
+
+        if (upw != null) {
+            return RabbitMQConnectionHolder.createConnection(env.expand(rHost),
                     rh.getRabbitMQPort(),
                     upw.getUsername(),
                     Secret.toString(upw.getPassword()),
-                    RabbitMQDefaults.AUTOMATIC_RECOVERY),
-                    filters);
-        } else {
-            broker = new TraceyRabbitMQBrokerImpl(new RabbitMQConnection(RabbitMQDefaults.HOST,
-                                                RabbitMQDefaults.PORT,
-                                                env.expand(RabbitMQDefaults.USERNAME),
-                                                env.expand(RabbitMQDefaults.PASSWORD),
-                                                RabbitMQDefaults.AUTOMATIC_RECOVERY),
-                                                filters);
+                    RabbitMQDefaults.AUTOMATIC_RECOVERY);
         }
-        return broker;
+        return RabbitMQConnectionHolder.createConnection(RabbitMQDefaults.HOST,
+                RabbitMQDefaults.PORT,
+                env.expand(RabbitMQDefaults.USERNAME),
+                env.expand(RabbitMQDefaults.PASSWORD),
+                RabbitMQDefaults.AUTOMATIC_RECOVERY);
     }
 
     @Override
     public void stop() {
+        LOG.log(Level.INFO, String.format("[%s] Stopping %s#%d trigger",
+                Thread.currentThread().getName(), getDescriptor().getDisplayName(), hashCode()));
+        if (broker != null) {
+            consumerTagCache.values().remove(consumerTag);
+            cancelConsumer(consumerTag);
```

Comment:
We are not closing channel. We just control consumers. If the connection was (auto)closed we create a new one in `configureBroker()`